src/encoded/static/components/matrix_target.js

```diff
@@ -0,0 +1,776 @@
+                    style: { backgroundColor, borderTop, borderRight },
+                };
+            }
+            return content;
+        });
+        const css = 'matrix__row-data';
+
+        return { rowContent, css };
+    });
+
+    dataTable.push(...rowData);
+
+    const rowKeys = [];
+    for (let i = 0; i < dataTable.length; i += 1) {
+        rowKeys.push(`key${i}`);
+    }
+
+    const matrixConfig = {
+        rows: dataTable,
+        rowKeys,
+        tableCss: 'matrix',
+    };
+
+    return matrixConfig;
+};
+
+const Spinner = ({ isSpinnerActive }) => (isSpinnerActive ?
+    <div className="communicating">
+        <div className="loading-spinner" />
+    </div> :
+    <div className="done">
+        <span>&nbsp;</span>
+    </div>);
+
+Spinner.propTypes = {
+    isSpinnerActive: PropTypes.bool,
+};
+
+Spinner.defaultProps = {
+    isSpinnerActive: false,
+};
+
+/**
+ * Target Matrix text filter.
+ *
+ *  Important to extend TextFilter because this class has functionality it lacks like
+ *  knowing when to clear text book via Pubsub subscription
+ *
+ * @class TargetMatrixTextFilter
+ * @extends {TextFilter}
+ */
+class TargetMatrixTextFilter extends TextFilter {
+    constructor() {
+        super();
+
+        this.handleChange = this.handleChange.bind(this);
+        this.clearSearch = this.clearSearch.bind(this);
+
+        this.state = { searchOption: 'biosample' };
+        this.searchBox = React.createRef();
+    }
+
+    componentDidMount() {
+        this.clearSearchPubSub = PubSub.subscribe(CLEAR_SEARCH_BOX_PUBSUB, this.clearSearch);
+    }
+
+    componentWillUnmount() {
+        PubSub.unsubscribe(this.clearSearchPubSub.unsubscribe);
+    }
+
+    clearSearch() {
+        if (this.searchBox && this.searchBox.current) {
+            this.searchBox.current.value = '';
+        }
+    }
+
+    onKeyDown(e) {
+        if (e.keyCode === 13) {
+            e.preventDefault();
+            PubSub.publish(SEARCH_PERFORMED_PUBSUB, {
+                text: e.target.value,
+                option: this.state.searchOption,
+            });
+        }
+    }
+
+    handleChange(e) {
+        this.setState({ searchOption: e.target.value });
+    }
+
+    render() {
+        return (
+            <div className="facet target-matrix-search">
+                <input
+                    type="search"
+                    className="search-query"
+                    placeholder="Enter search term(s)"
+                    defaultValue={this.getValue(this.props)}
+                    onKeyDown={this.onKeyDown}
+                    data-test="filter-search-box"
+                    ref={this.searchBox}
+                />
+                <select name="searchOption" onChange={this.handleChange}>
+                    <option value="biosample">Biosample</option>
+                    <option value="target">Target</option>
+                </select>
+            </div>
+        );
+    }
+}
```

Comment:
The React team discourages the use of component inheritance, and I think that must be especially true with the emphasis on functional components with hooks. https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance

Reply:
@zoldello explained to me the issue and I now agree that the best thing for now is to keep this as is. I wrote ENCD-5067 to address this in the future.

Reply:
Moved the inheritance. TabPanel may be revisited and used in the future instead of TargetTabPanel